[cutlass backend] compile and link for .so files

henrylhtsang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #155876

Differential Revision: [D76482736](https://our.internmc.facebook.com/intern/diff/D76482736/)

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

henrylhtsang committed:
[cutlass backend] compile and link for .so files



Differential Revision: [D76482736](https://our.internmc.facebook.com/intern/diff/D76482736/)

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] compile and link for .so files"


Differential Revision: [D76482736](https://our.internmc.facebook.com/intern/diff/D76482736/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang committed:
Update on "[cutlass backend] compile and link for .so files"


Differential Revision: [D76482736](https://our.internmc.facebook.com/intern/diff/D76482736/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

ColinPeppler commented:
is it straightforward to add a test case?

coconutruben commented:
code looks fine, but yeah it would be good to have 2 test cases in here
- `.so` being compiled
- `.o` being compiled directly (doesn't loop, etc)
but the logic is quite nice and elegant to leverage the underlying work

coconutruben commented:
are there existing tests that just exercise the compile path? or is it all wrapped in larger functionality (export + loading, etc)? I might be nice to have an explicit test that just compiles a `.so` and makes sure a `.o` is still lying around afterwards

henrylhtsang commented:
> are there existing tests that just exercise the compile path? or is it all wrapped in larger functionality (export + loading, etc)? I might be nice to have an explicit test that just compiles a `.so` and makes sure a `.o` is still lying around afterwards

they are used in larger functionality.

I guess I don't get the point of making a test that just call into this function. This just seems like a pretty fragile test that doesn't test if something is working, but instead of something goes through the path you expect it to.

henrylhtsang committed:
Update on "[cutlass backend] compile and link for .so files"


Differential Revision: [D76482736](https://our.internmc.facebook.com/intern/diff/D76482736/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang commented:
looks like there are some codecache tests

henrylhtsang committed:
Update on "[cutlass backend] compile and link for .so files"


Differential Revision: [D76482736](https://our.internmc.facebook.com/intern/diff/D76482736/)

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

henrylhtsang commented:
@pytorchbot merge
